src/solver/conv_mlir_igemm_fwd_xdlops.cpp

```diff
@@ -139,14 +151,35 @@ bool PerformanceConvMlirIgemmXdlops::SetNextValue(const ConvolutionContext& /*co
             break;
         if(!NextTwoPower<16, 256>(GemmNPerBlock))
             break;
-        if(!NextTwoPower<1, 8>(GemmKPerBlock))
+        if(!NextTwoPower<4, 8>(GemmKPerBlock))
             break;
         if(!NextTwoPower<4, 128>(GemmMPerWave))
             break;
         if(!NextTwoPower<16, 128>(GemmNPerWave))
             break;
-        if(!NextTwoPower<4, 8>(GemmKPACKSize))
-            break;
+        if(config.IsFp32())
+        {
+            if(config.direction.IsForward())
+            {
+                if(!NextTwoPower<1, 4>(GemmKPACKSize))
```

Comment:
What about 8?

Reply:
That's right I didn't notice it is under fp32 condition, but what about 2? I do notice that 2 was previously not included but we can potentially include in the future?

Reply:
`NextTwoPower<1,4>` covers [1, 2, 4].